### PR TITLE
Revised Issue #15 NoConnection Interface and Callback to Fragments + General SnackBar Utility

### PR DESCRIPTION
**Pull request by laavanyebahl**
Issue no #15 Revised
- API exception checked while syncing
- Exception type is checked
- Interface for NetworkCallBack
- Callback implemented in Fragment View Pager
- General Reusable SnackBar Utility

ff

### CODE REVIEW

File: api/src/main/java/org/hisp/dhis/android/dashboard/api/utils/NetworkUtils.java [hunk 1]
```diff
@@ -45,6 +45,15 @@ private NetworkUtils() {
         // no instances
     }
 
+    public interface NetworkCallBack {
+         void networkCallback();
+    }
+    static NetworkCallBack networkCallBack;
```
Comment (arazabishov) [inline api/src/main/java/org/hisp/dhis/android/dashboard/api/utils/NetworkUtils.java:51]: Keeping static reference to callback can cause a memory leak.

File: app/src/main/java/org/hisp/dhis/android/dashboard/ui/fragments/dashboard/DashboardViewPagerFragment.java [hunk 2]
```diff
@@ -265,6 +267,13 @@ public void onResponseReceived(NetworkJob.NetworkJobResult<?> result) {
         }
     }
 
+    // Callback for No internet connection
+    NetworkUtils networkUtils = new NetworkUtils(this);
```
Comment (arazabishov) [inline app/src/main/java/org/hisp/dhis/android/dashboard/ui/fragments/dashboard/DashboardViewPagerFragment.java:271]: Here you are implicitly capturing reference to Fragment through NetworkCallback implementation in class. It means, Activity to which Fragment is attached, will be leaked through static reference to callback in NetworkUtils.

**Comment by arazabishov** [comment]
@laa-1-yay Thanks for making effort to improve previous pull-request, but there are still some problems in your code. Please, see comments attached to pull-request.

**Comment by laavanyebahl** [comment]
@ArazAbishov Thanks for reviewing the code. I'll make changes again as per the suggestions and submit a pr again.

**Pull request closed without merging**